nucliadb/src/nucliadb/search/search/graph_strategy.py

```diff
@@ -0,0 +1,633 @@
+                break
+
+    # Get the text blocks of the paragraphs that contain the top relations
+    with metrics.time("graph_strat_build_response"):
+        paragraph_ids = {
+            r.metadata.paragraph_id
+            for rel in relations.entities.values()
+            for r in rel.related_to
+            if r.metadata and r.metadata.paragraph_id
+        }
+        find_request = find_request_from_ask_request(item, query)
+        query_parser, rank_fusion, reranker = await query_parser_from_find_request(
+            kbid, find_request, generative_model=generative_model
+        )
+        find_results = await build_graph_response(
+            paragraph_ids,
+            kbid=kbid,
+            query=query,
+            final_relations=relations,
+            top_k=graph_strategy.top_k,
+            reranker=reranker,
+            show=find_request.show,
+            extracted=find_request.extracted,
+            field_type_filter=find_request.field_type_filter,
+        )
+
+    return find_results, query_parser
+
+
+async def fuzzy_search_entities(
+    kbid: str,
+    query: str,
+    show: list[ResourceProperties],
+    field_type_filter: list[FieldTypeName],
+    range_creation_start: Optional[datetime] = None,
+    range_creation_end: Optional[datetime] = None,
+    range_modification_start: Optional[datetime] = None,
+    range_modification_end: Optional[datetime] = None,
+    target_shard_replicas: Optional[list[str]] = None,
+) -> KnowledgeboxSuggestResults:
+    """Fuzzy find entities in KB given a query using the same methodology as /suggest, but split by words."""
+
+    base_request = nodereader_pb2.SuggestRequest(
+        body="", features=[nodereader_pb2.SuggestFeatures.ENTITIES]
+    )
+    if range_creation_start is not None:
+        base_request.timestamps.from_created.FromDatetime(range_creation_start)
+    if range_creation_end is not None:
+        base_request.timestamps.to_created.FromDatetime(range_creation_end)
+    if range_modification_start is not None:
+        base_request.timestamps.from_modified.FromDatetime(range_modification_start)
+    if range_modification_end is not None:
+        base_request.timestamps.to_modified.FromDatetime(range_modification_end)
+
+    tasks = []
+    # XXX: Splitting by words is not ideal, in the future, modify suggest to better handle this
+    for word in query.split():
+        if len(word) <= 3:
+            continue
+        request = nodereader_pb2.SuggestRequest()
+        request.CopyFrom(base_request)
+        request.body = word
+        tasks.append(
+            node_query(kbid, Method.SUGGEST, request, target_shard_replicas=target_shard_replicas)
+        )
+
+    try:
+        results_raw = await asyncio.gather(*tasks)
+        return await merge_suggest_results(
+            [item for r in results_raw for item in r[0]],
+            kbid=kbid,
+            show=show,
+            field_type_filter=field_type_filter,
```

Comment:
you can remove these two parameters, they are not being used by the inner function.

https://github.com/nuclia/nucliadb/blob/main/nucliadb/src/nucliadb/search/search/merge.py#L588

feel free to remove them from the original function aswell as this will simplify your code